lms/views/oauth.py

```diff
@@ -19,8 +19,8 @@ def build_canvas_token_url(lms_url):
 def canvas_oauth_callback(request):
     """Route to handle content item selection oauth response."""
     state = request.params['state']
-    oauth_state = find_by_state(request.db, state)
-    lti_params = json.loads(oauth_state.lti_params)
+
+    lti_params = lti_params_for(request)
```

Comment:
This will now raise `HTTPBadRequest` if the OAuth redirect request is invalid (it has no corresponding stashed "state" / LTI params in the DB). Before it would have continued and just crashed on the next line.

Reply:
Are there situations in which this could be raised not from a bad request but from something internal? `400` is definitely better than `500` here, and probably appropriate, just a chance to stop and reflect (we don't want to accuse the client of a bad request if it's something we've actually screwed up).